1-JavaScript/Week3/homework/ex4-shoppingCart.js

```diff
@@ -20,10 +20,23 @@ you have more than 3 items in your shopping cart the first item gets taken out.
    argument. Use `console.log` to display the results.
 -----------------------------------------------------------------------------*/
 const shoppingCart = ['bananas', 'milk'];
+function addToShoppingCart(groceryItem) {
+  let list=''
+  shoppingCart.push(groceryItem);
+  if (shoppingCart.length>3){
+    shoppingCart.shift();
+  }
+  for (const n of shoppingCart){
+    list+=', '+n;
+  }
+  const message = `You bought${list}!`;
+  return message;
+  
+  
+  }
```

Comment:
the solution works fine but there will be some simpler approaches. Here you have one of them:

```suggestion
function addToShoppingCart(groceryItem) {
  shoppingCart.push(groceryItem);
  
  if (shoppingCart.length>3){
    shoppingCart.shift();
  }

  const message = `You bought ${shoppingCart.join(', ')}!`;
  
  return message;
}
```

The shoppingCart.join method converts an array into a string. AS you can see in this example, join allows you to specify a sperator. For this exercise, we would like to separate each item from the array using a comma and a space.

Here you can find more information about the join method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join